### PR TITLE
include error messages in predictions (fixes #668)

### LINKED ISSUE #668: The error message in a FailureModel is not stored for predictions?
```r
configureMlr(on.learner.error = "warn")

lrn = makeLearner("classif.knn")
# randomForest stores the error messages
# lrn = makeLearner("classif.randomForest")
lrn$properties = c(lrn$properties, c("missings"))

task = sonar.task
task$env$data$V1[1:2] = NA
r = crossval(lrn, task)
r$err.msgs
```

```
   iter train predict
1     1  <NA>    <NA>
2     2  <NA>    <NA>
3     3  <NA>    <NA>
4     4  <NA>    <NA>
5     5  <NA>    <NA>
6     6  <NA>    <NA>
7     7  <NA>    <NA>
8     8  <NA>    <NA>
9     9  <NA>    <NA>
10   10  <NA>    <NA>
```

Shouldn't the `predict` column contain the error messages? Using `lrn = makeLearner("classif.randomForest")` for example stores an error message in the `train` column.

### PR DESCRIPTION
**Pull request by larskotthoff**

### CODE REVIEW

File: R/resample.R [hunk 1]
```diff
@@ -124,14 +124,18 @@ doResampleIteration = function(learner, task, rin, i, measures, weights, model,
...
   } else if (pp == "test") {
     pred.test = predict(m, task, subset = test.i)
+    if (!is.null(pred.test$error)) err.msgs[2L] = pred.test$error
     ms.test = vnapply(measures, function(pm) performance(task = task, model = m, pred = pred.test, measures = pm))
   } else { # "both"
     pred.train = predict(m, task, subset = train.i)
+    if (!is.null(pred.train$error)) err.msgs[2L] = pred.train$error
     ms.train = vnapply(measures, function(pm) performance(task = task, model = m, pred = pred.train, measures = pm))
     pred.test = predict(m, task, subset = test.i)
+    if (!is.null(pred.test$error)) err.msgs[2L] = paste(err.msgs[2L], pred.test$error)
```
Comment (mllg) [inline R/resample.R]: Is this `NULL` or `NA_character_`?

Comment (larskotthoff) [inline R/resample.R]: Should be NULL as far as I can see.

**Comment by larskotthoff** [comment]
Not sure what's going on with the test failures; I can't reproduce this locally.

Should be ready to be merged.

**Comment by berndbischl** [comment]
> Not sure what's going on with the test failures; I can't reproduce this locally.
> 
> Should be ready to be merged.

both sentences contradict each other. i mean, come on....
the error is also reproducible locally and seems to have something todo with the changes we do wrt prediction!

**Comment by berndbischl** [comment]
the error is, as one can clearly see...., in the test_classif_svm. there we count the number of elements in the Prediction-list-object. which is now 6, not 5.

but the whole test seems rather stupid. there is really no reason to do this in test_classif_svm

also this is a change to mlr code. should be looked at by multiple people. 
i will review later.

@ja-thomas @jakob-r @schiffner 
care to take a look here?

**Comment by larskotthoff** [comment]
Should be fixed now.

**Comment by berndbischl** [comment]
in your test. at the end: 

configureMlr(on.learner.error = "stop")
configureMlr(on.learner.warning = "warn")

this is bad style, pls dont do this. reset the optoins to what they were, before you set them to "quiet". not what you guess they should be.

**Comment by berndbischl** [comment]
regarding you code in resample.R:
these lines:
 if (!is.null(pred.test$error)) err.msgs[2L] = pred.test$error

i have the same question as @mllg 

you have defined "err msg not present" in Prediction-object with NA_char?
now for what reason do you check against NULL?
also you have not documented this fact in Prediction.R

**Comment by larskotthoff** [comment]
Fixed.

**Comment by berndbischl** [comment]
thx!

i guess this is OK now.

**Pull request merged**